doc/ampi/manual.rst

```diff
@@ -519,53 +519,227 @@ provided that it takes place *after* ``MPI_Init``.
 Global Variable Privatization
 -----------------------------
 
-For the before-mentioned benefits to be effective, one needs to map
-multiple user-level threads onto each processor. Traditional MPI
-programs assume that the entire processor is allocated to themselves,
-and that only one thread of control exists within the process’s address
-space. So, they may safely use global and static variables in the
-program. However, global and static variables are problematic for
+For the aforementioned benefits to be effective, AMPI needs to map
+multiple user-level threads onto each OS process. Traditional MPI
+programs assume that each rank has an entire OS process to itself,
+and that only one thread of control exists within its address space.
+This allows them to safely use global and static variables in their
+code. However, global and static variables are problematic for
 multi-threaded environments such as AMPI or OpenMP. This is because
-there is a single instance of those variables so they will be shared
-among different threads in the single address space, so if programmers
-are not careful a wrong result may be produced by the program. Figure
+there is a single instance of those variables, so they will be shared
+among different threads in the single address space, and this could
+lead to the program producing an incorrect result or crashing. Figure
 :numref:`fig_global` shows an example of a multi-threaded
 application with two threads in a single process. :math:`var` is a
 global or static variable in this example. Thread 1 assigns a value to
 it, then it gets blocked for communication and another thread can
-continue. Thereby, thread 2 is scheduled next and accesses :math:`var`
+continue. Then, thread 2 is scheduled next and accesses :math:`var`,
```

Comment:
Is this example (and the figure below) really helpful or necessary?

Reply:
I think _some_ example is important to have here, for illustration. I don't know if the figure is the best approach. Instead we could always use this simpler example from presentations I have given:

```c
int rank_global;

void print_ranks(void)
{
  MPI_Comm_rank(MPI_COMM_WORLD, &rank_global);

  MPI_Barrier(MPI_COMM_WORLD);

  printf("rank: %d\n", rank_global);
}
```

Reply:
> I think _some_ example is important to have here, for illustration. I don't know if the figure is the best approach. Instead we could always use this simpler example from presentations I have given:
> 
> ```c
> int rank_global;
> 
> void print_ranks(void)
> {
>   MPI_Comm_rank(MPI_COMM_WORLD, &rank_global);
> 
>   MPI_Barrier(MPI_COMM_WORLD);
> 
>   printf("rank: %d\n", rank_global);
> }
> ```

I think this example (+ 1 phrase of explanation or so) works much better.